### ISSUE TITLE
Duplicated Synchronization with Gradient Clipping?

### ISSUE THREAD

**Issue by ildoonet** [horovod/horovod#1099]
**Environment:**
1. Framework: pytorch
2. Framework version: 1.0
3. Horovod version: latest
4. MPI version: -
5. CUDA version: -
6. NCCL version: -
7. Python version: 3.6
8. OS and version: Ubuntu 16.04

https://github.com/horovod/horovod/blob/master/horovod/torch/__init__.py#L174

If you see this documentation, when using gradient clipping, there is duplicated(twice) synchronization. Once is when 'synchronize' is called and second one is when 'step' is called.

After I followed this documentation with gradient clipping, training speed is slower.

Any solusions?

**Comment by alsrgv** [comment]
@ildoonet, thanks for raising this!  This regression has been introduced by #597.

While we're thinking about a proper solution, can you specify `optimizer._requires_update = set()` after `DistributedOptimizer` wraps original optimizer, [like this](https://gist.github.com/alsrgv/0713add50fe49a409316832a31612dde)?

**Comment by ildoonet** [comment]
@alsrgv Thanks for the tip. I will try with it and wait for the proper solutions.

**Comment by alsrgv** [comment]
@ildoonet, the fix was merged into master.  You can reinstall Horovod from master (or wait a bit for 0.16.3), and use `.step(synchronize=False)`, as new documentation prescribes.
